charmcraft.yaml

```diff
@@ -14,4 +14,4 @@ parts:
   charm:
     charm-binary-python-packages:
       - "pydantic>=2"
-      - "opentelemetry-exporter-otlp-proto-grpc==1.17.0"
```

Comment:
whether this dependency lives here or in requirements.txt, charmcraft will pick it up regardless and packing time will be the same.

(we usually put it in both though, otherwise tox will not automatically recognize it as a dependency and we need to manually add it to all envs)

Reply:
the change is compatible with requirer charms that use grpc, but it will require them to ALSO install the otlp_http lib.

ATM a requirer charm obtains from tempo all endpoints (not just the ones it needs: see #33 )

So the upgrade story for  charm_tracing v0 --> v1 is going to be:

(for tracing-requirer charms)
- bump charm lib to v1
- replace the grpc endpoint url to the otlp_http one in the charm method `@trace` is pointing to (using the existing `TracingRequirer.otlp_http_endpoint()` method)
- instead of returning a ssl cert, return a path to it from the other method (if using tls)
- profit

Reply:
so if a charm was using otlp_grpc to send traces (and to no other purpose), they can drop the python dependency and add one to otlp_http instead.

If they also need otlp_grpc for their own thing, that's not our problem.

Just like before this change, if a charm depended on otlp_http for their own business, we were forcing them to also depend on otlp_grpc.